### PR TITLE
gsl: build against external cblas

### PR DESCRIPTION
**Pull request by junghans**
@mgorny is the original author of the patch, but it has been used in gentoo for a while.

### CODE REVIEW

File: var/spack/repos/builtin/packages/gsl/package.py [hunk 1]
```diff
@@ -24,3 +24,31 @@ class Gsl(AutotoolsPackage, GNUMirrorPackage):
...
+    depends_on('m4',       type='build', when='+external-cblas')
+    depends_on('autoconf', type='build', when='+external-cblas')
+    depends_on('automake', type='build', when='+external-cblas')
+    depends_on('libtool',  type='build', when='+external-cblas')
+    depends_on('blas', when='+external-cblas')
+
+    @property
+    def force_autoreconf(self):
+        # The external cblas patch touches configure
+        return self.spec.satisfies('+external-cblas')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gsl/package.py]: Alternatively, we could use the patch to regenerate configure, then develop our own patch to configure. That way we don't need to build the extra dependencies. Up to you.

File: var/spack/repos/builtin/packages/gsl/package.py [hunk 2]
```diff
@@ -24,3 +24,31 @@ class Gsl(AutotoolsPackage, GNUMirrorPackage):
...
+
+    def configure_args(self):
+        configure_args = []
+        if self.spec.satisfies('+external-cblas'):
+            configure_args.append('--with-external-cblas')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gsl/package.py]: Can you explicitly disable this in an else clause?

File: var/spack/repos/builtin/packages/gsl/package.py (hunk 1 above)
Comment (junghans) [reply var/spack/repos/builtin/packages/gsl/package.py]: I usually don’t like that approach as patches against configure quickly become very messy and big if you use exactly the same autotools, plus we might end up with different patches for each version of gsl.

File: var/spack/repos/builtin/packages/gsl/package.py (hunk 2 above)
Comment (junghans) [reply var/spack/repos/builtin/packages/gsl/package.py]: Well, the option does exist in unpatched/disabled case.

**Pull request merged**